design/baremetal-operator/bmh_non-bootable_iso.md

```diff
@@ -0,0 +1,142 @@
+# Add non-bootable iso attach API to BareMetalHost
+
+## Status
+
+Implementable
+
+## Summary
+
+Add a new Custom Resource(CR), `DataImage`, to the cluster to attach a
+generic non-bootable ISO (a CD "ISO 9660" image) image to a provisioned
+or externallyProvisioned BMH using Ironic.
+
+## Motivation
+
+In certain scenarios it is desirable to attach a non-bootable iso image
+to a system after provisioning has finished, for example:
+
+A configuration ISO can be used to provide first-boot configuration to a host
+deployed from a standard image in advance. An operator may have a pool of
+powered off hosts with operating system already provisioned. Then, adding
+a host to the cluster boils down to powering it on with a configuration ISO provided.
+
+### Goals
+
+Expose the Ironic API to attach non-bootable iso images via Metal3.
+
+### Non-Goals
+
+* Supporting non-ISO images i.e. media types other than CD/DVD
+* Supporting drivers other than Redfish and its derived drivers.
+
+## Proposal
+
+### BMO Proposal
+
+Add a new Custom Resource(CR), `DataImage` (with same `name` and `namespace` as the BMH), which contains the spec containing details of the
+non-bootable iso image, example :
+
+#### Design spec
+
+```yaml
+apiVersion: metal3.io/v1alpha1
+kind: DataImage
+metadata:
+  name: metal-worker-0
+  namespace: metal-cluster
+spec:
+  url: http://1.2.3.4/image.iso
+```
+
+We will use ownerReferences in the `DataImage` CR to link it to the corresponding `BareMetalHost` object.
+
+Since we are using a CR, a user can create RBAC policies that doesn't allow unauthorized users
+to attach a non-bootable iso image to hosts.
+
+We may consider adding support for HTTP credentials and TLS settings in the future
+but it's outside the scope of this proposal.
+
+**Note** : As part of this proposal, the dataImage will be attached after the node
+has been Provisioned or ExternallyProvisioned.
+
+### Ironic Support
+
+The [proposal](https://bugs.launchpad.net/ironic/+bug/2033288) for exposing this functionaility in Ironic has been
+accepted by the upstream Ironic community.
+
+The [Ironic implementation](https://review.opendev.org/c/openstack/ironic/+/894918) based on the above proposal has been completed.
+
+Other than that, we just need the implementation for driver specific support,
+which in this proposal is Redfish.
+
+## Design Details
+
+Create a new Custom Resource(CR), `DataImage`, with spec field `Url` and the same `Name`
+and `Namespace` as the `BareMetalHost` where we are attaching the image.
+
+To attach an ISO, first the `DataImage` CR object is created and then it is attached to the
+BMH on its next reboot/power on.
+
+The `BareMetalHostStatus` will cache the image details including the attachment
```

Comment:
This part still references the BMH status.

Reply:
Fixed